### PR TITLE
Added libe57format port

### PR DESCRIPTION
**Pull request by joseangelmt**
**Describe the pull request**

A library to provide read & write support for the E57 file format.

- #### What does your PR fix?  
  Adds a new port ([libe57format](https://github.com/asmaloney/libE57Format))

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libe57format/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO asmaloney/libE57Format
+    REF v2.2.0
+    SHA512 7b788efce2efdbfba83d4e3ab5c4884b3d85b5e44c510954f9200db26fb97c92f4e33d04209c0647fc39c3af081fc09d1ee6a79857b1980a0e1e43586f78bd0c
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
```
Comment (mathisloge) [inline ports/libe57format/portfile.cmake:9]: `vcpkg_configure_cmake` is deprecated, see https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md 

You should use `vcpkg_cmake_configure` and add 
```cmake
{
      "name": "vcpkg-cmake",
      "host": true
    }
``` to the dependencies.

File: ports/libe57format/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    SHA512 7b788efce2efdbfba83d4e3ab5c4884b3d85b5e44c510954f9200db26fb97c92f4e33d04209c0647fc39c3af081fc09d1ee6a79857b1980a0e1e43586f78bd0c
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+)
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH "lib/cmake/E57Format")
```
Comment (mathisloge) [inline ports/libe57format/portfile.cmake:14]: same as above, but use `vcpkg_cmake_config_fixup`

and add 
```json
{
      "name": "vcpkg-cmake-config",
      "host": true
    },
``` 
to the dependencies

**Review by mathisloge — CHANGES_REQUESTED** [review]
see https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md and https://github.com/Microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_cmake.md for deprecation notice

File: ports/libe57format/portfile.cmake (hunk 1 above)
Comment (joseangelmt) [reply ports/libe57format/portfile.cmake]: Thank you Mathis,

I already changed both: vcpkg_configure_cmake and cpkg_fixup_cmake_targets.
What confused me was the [Packaging Github Repos Example: libogg](https://vcpkg.io/en/docs/examples/packaging-github-repos.html) documentation.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**